File: main/java/com/marsrovers/control/Command.java
package com.marsrovers.control;

import java.util.List;

import static java.util.Arrays.asList;
import static java.util.Collections.emptyList;

public class Command {

    private static final String VALID_COMMAND_REGEX = "[LMR]+";
    private static final String MOVE_COMMAND_REGEX= "[M]+";
    private static final String EMPTY_REGEX = "";

    private final String commandString;

    public Command(String commandString) {
        this.commandString = commandString;
    }

    private boolean isExecutable() {
        return commandString != null && commandString.matches(VALID_COMMAND_REGEX);
    }

    boolean containsMove() {
        return commandString != null && commandString.matches(MOVE_COMMAND_REGEX);
    }

    List<String> getMoveList() {
        return isExecutable()
                ? asList(commandString.split(EMPTY_REGEX))
                : emptyList();
    }

}

File: main/java/com/marsrovers/control/Command.kt
package com.marsrovers.control

import java.util.Arrays.asList
import java.util.Collections.emptyList

class Command(private val commandString: String) {

    private fun isExecutable(): Boolean = commandString.matches(VALID_COMMAND_REGEX.toRegex())

    fun containsMove(): Boolean = commandString.matches(MOVE_COMMAND_REGEX.toRegex())

    internal val moveList: List<String>
        get() = if (isExecutable())
            asList(*commandString.split(EMPTY_REGEX.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray())
        else
            emptyList<String>()

    companion object {

        private val VALID_COMMAND_REGEX = "[LMR]+"
        private val MOVE_COMMAND_REGEX = "[M]+"
        private val EMPTY_REGEX = ""
    }

}
